addons/analytic/models/analytic_line.py

```diff
@@ -107,26 +105,7 @@ def _search_auto_account(self, operator, value):
 
     def _get_view(self, view_id=None, view_type='form', **options):
         arch, view = super()._get_view(view_id, view_type, **options)
-        if self.env['account.analytic.plan'].check_access_rights('read', raise_exception=False):
-            project_plan, other_plans = self.env['account.analytic.plan']._get_all_plans()
-
-            # Find main account nodes
-            account_node = arch.find('.//field[@name="account_id"]')
-            account_filter_node = arch.find('.//filter[@name="account_id"]')
-
-            # Force domain on main account node as the fields_get doesn't do the trick
-            if account_node is not None and view_type == 'search':
-                account_node.set('domain', repr([('plan_id', 'child_of', project_plan.id)]))
-
-            # If there is a main node, append the ones for other plans
-            if account_node is not None or account_filter_node is not None:
-                for plan in other_plans[::-1]:
-                    fname = plan._column_name()
-                    if account_node is not None:
-                        account_node.addnext(E.field(name=fname, domain=f"[('plan_id', 'child_of', {plan.id})]", optional="show"))
-                    if account_filter_node is not None:
-                        account_filter_node.addnext(E.filter(name=fname, context=f"{{'group_by': '{fname}'}}"))
-        return arch, view
+        return self.env['account.analytic.plan']._view_add_fields(arch, view, view_type)
 
     @api.model
     def fields_get(self, allfields=None, attributes=None):
```

Comment:
You also need to add this everywhere you call `_view_add_fields`
Otherwise, the domain don't be everywhere as expected
![image](https://github.com/odoo/odoo/assets/44770049/2d12200d-4d81-4c43-9d27-a95faff6cd76)


Maybe a mixin implementing `fields_get` and `_get_view` might be cleaner/more robust

Reply:
@william-andre As I know this is targeted to SaaS-17.3 and now we only few hours.
so we may improve afterwards before V18.
what do you think?

Reply:
Start working on this already, there is still stuff coming in enterprise.

Reply:
okay, so you are working on this or just reviewing it?

Reply:
I'm reviewing it